sdk/scheduler/src/main/java/com/mesosphere/sdk/state/PersistentOperationRecorder.java

```diff
@@ -34,7 +36,13 @@ public void record(Operation operation, Offer offer) throws Exception {
     private void recordTasks(List<Protos.TaskInfo> taskInfos) throws StateStoreException {
         logger.info(String.format("Recording %d updated TaskInfos/TaskStatuses:", taskInfos.size()));
         List<Protos.TaskStatus> taskStatuses = new ArrayList<>();
-        for (Protos.TaskInfo taskInfo : taskInfos) {
+        Collection<Protos.TaskInfo> unpackedTaskInfos;
+        try {
+            unpackedTaskInfos = CommonTaskUtils.unpackTaskInfos(taskInfos);
```

Comment:
Instead of unpacking from `OfferRecommendation.getOffer()`, how about passing the original TaskInfo directly into `LaunchOfferRecommendation`, and then performing the packing inside of there? Then `PersistentOperationRecorder` can do something like:

```
if (operation instanceof LaunchOfferRecommendation) {
  LaunchOfferRecommendation lro = (LaunchOfferRecommendation) operation;
  recordTasks(lro.getOriginalTaskInfos());
}
```

This should make it possible to ensure that only the executor is calling `unpackTaskInfos()`